### ISSUE TITLE
Installation failure

### ISSUE THREAD

**Issue by herbzhao** [stablecabal/gyre-installer#3]
Quite a few problems here and there, for example the parent folder cannot have space.

Also when finally installed, and press run.cmd, the below error complains:

hinters-uniformer-small. Failed attempts:
gyre               |   - Path 'weights\hinters-uniformer-small' isn't a directory

Error when trying to load weights from
\.cache\huggingface\hub\models--halffried--gyre_hinters\snapshots\50822d9a0347f2ab0fa1a1c7b48addc8ab5c998a\uniformer_small/. DLL load failed while importing _ext: The specified module could not be found.

**Comment by hafriedlander** [comment]
Hi,

I haven't seen that error before, and I'm not really sure what could be causing it.

Some of the usual problems I've seen during installation
- Running out of drive space
- An antivirus causing problems (removing models after download)

The installer currently isn't very good at detecting errors during installation, I'm improving that now. I'll update this ticket when a new version is ready.

**Comment by hafriedlander** [comment]
Just an update, there is another user who has seen this issue now. I believe it is an issue with the "mmcv" library. I will update this ticket once I have replicated the issue.

In the meantime I have released a newer installer that might help. Please start again, using https://github.com/stablecabal/gyre-installer/archive/refs/tags/bundle-v2.0.1.zip

**Comment by herbzhao** [comment]
Nope. tried again, still complain. Maybe you should do a docker install somehow to pack cuda etc in one place? 

Error when trying to load weights from ....\.cache\huggingface\hub\models--halffried--gyre_hinters\snapshots\50822d9a0347f2ab0fa1a1c7b48addc8ab5c998a\uniformer_small/. DLL load failed while importing _ext: The specified module could not be found.

I have removed the cache manually but same outcome.

**Comment by hafriedlander** [comment]
There is a docker image, https://gyre.ai/docs/install/docker

**Comment by herbzhao** [comment]
OK this finally got it working. If you update the future, do i need to restart the entire downloading process each time? it is taking a while..

**Comment by hafriedlander** [comment]
Using gyre-installer: no, all the models are cached, and the updater only updates the changed files.

Using docker: no, as long as you've mapped the /huggingface folder. (The instructions on that page above do this with this argument: `-v $HOME/.cache/huggingface:/huggingface`).

Did you get it working with gyre-installer, or the docker?

**Comment by herbzhao** [comment]
> Using gyre-installer: no, all the models are cached, and the updater only updates the changed files.
> 
> Using docker: no, as long as you've mapped the /huggingface folder. (The instructions on that page above do this with this argument: `-v $HOME/.cache/huggingface:/huggingface`).
> 
> Did you get it working with gyre-installer, or the docker?

Hi, i get the docker working, but the memory usage is extreme. I drew 2 images and it immediately sucked up entire 32gb ram without giving back?

**Comment by hafriedlander** [comment]
Hi. RAM (CPU) or VRAM (GPU)? And what do you mean by "giving back", are you getting Out-Of-Memory errors, or do you mean it's not releasing RAM after a generation?

If the later, then that's sort of by design. It's a server, so it doesn't load everything every time - that would be much too slow. In particular, all the models are loaded into RAM (not VRAM) on startup. It should work on 16GB ram though.

You can add `SD_ENV=dev` as an environment variable and it will put RAM and VRAM usage metrics into the logs. Posting those will help tell if there's something unusual going on.
